src/main/java/seedu/contax/storage/CsvManager.java

```diff
@@ -0,0 +1,93 @@
+package seedu.contax.storage;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Set;
+
+import seedu.contax.logic.commands.CommandResult;
+import seedu.contax.logic.commands.ImportCsvCommand;
+import seedu.contax.logic.parser.ParserUtil;
+import seedu.contax.logic.parser.exceptions.ParseException;
+import seedu.contax.model.IndexedCsvFile;
+import seedu.contax.model.Model;
+import seedu.contax.model.person.Address;
+import seedu.contax.model.person.Email;
+import seedu.contax.model.person.Name;
+import seedu.contax.model.person.Person;
+import seedu.contax.model.person.Phone;
+import seedu.contax.model.person.exceptions.DuplicatePersonException;
+import seedu.contax.model.tag.Tag;
+
+public class CsvManager {
+    private Model model;
+
+    /**
+     * Creates a CsvManager to import to specified {@code model}
+     */
+    public CsvManager(Model model) {
+        this.model = model;
+    }
+
+    /**
+     * Imports a given {@code IndexedCSvFile} to a {@code model}
+     * @param csvFile File to be imported
+     * @return CommandResult upon success
+     * @throws IOException if reading from file fails
+     */
+    public CommandResult importCsv(IndexedCsvFile csvFile) throws IOException {
+        BufferedReader importedCsv = new BufferedReader(new FileReader(csvFile.getFilePath()));
+        String line = "";
+        ArrayList<Integer> skippedLines = new ArrayList<>();
+        int lineCounter = 0;
+        //skip first line as it contains headers
+        importedCsv.readLine();
+
+        while ((line = importedCsv.readLine()) != null) {
+            lineCounter++;
+            String[] importedPerson = line.split(",");
+            try {
+                Person toAddPerson = personParser(importedPerson, csvFile);
+                model.addPerson(toAddPerson);
+                Set<Tag> tags = toAddPerson.getTags();
+                for (Tag tag : tags) {
+                    if (!model.hasTag(tag)) {
+                        model.addTag(tag);
+                    }
+                }
+            } catch (ParseException | DuplicatePersonException e) {
+                skippedLines.add(lineCounter);
+                continue;
+            }
+        }
+        return outputStringBuilder(skippedLines);
+    }
```

Comment:
I suggest separating the CSV logic from the import logic using a `Function`.

```suggestion
    private BiFunction<Integer, String[], Boolean> lineAction;

    // Return list of skipped lines
    public List<Integer> importCsv(IndexedCsvFile csvFile) throws IOException {
        BufferedReader importedCsv = new BufferedReader(new FileReader(csvFile.getFilePath()));
        String line = "";
        ArrayList<Integer> skippedLines = new ArrayList<>();
        int lineCounter = 0;
        //skip first line as it contains headers
        importedCsv.readLine();

        while ((line = importedCsv.readLine()) != null) {
            lineCounter++;
            String[] importedPerson = line.split(",");
            boolean successful = lineAction.apply(lineCounter, importedPerson);
            if (!successful) {
                skippedLines.add(lineCounter);
            }
        }
        return skippedLines;
    }
```

Reply:
thanks for the suggestion! resolved with next pr, do lmk if still got any issues